pkg/apis/core/validation/validation.go

```diff
@@ -2306,6 +2306,22 @@ func ValidateVolumeMounts(mounts []core.VolumeMount, voldevices map[string]strin
 			}
 		}
 
+		if len(mnt.SubPathExpr) > 0 {
+			if !utilfeature.DefaultFeatureGate.Enabled(features.VolumeSubpath) {
```

Comment:
We recently found there are [issues](https://github.com/kubernetes/kubernetes/issues/70270) with this validation in downgrade scenarios.  After downgrading or disabling the feature, you won't be able to delete your pods because it will fail this validation.

My suggested approach is to not check for feature gates here.  Instead, in [Pod PrepareForUpdate](https://sourcegraph.com/github.com/kubernetes/kubernetes@0914272a42bddfce24d77403aee7c6ef714f6b3a/-/blob/pkg/registry/core/pod/strategy.go#L79), we should:
* allow subpath to continue being set if the old object already used it
* disallow changing subpath

And in PrepareForCreate:
* drop the field if the feature gate is not set

cc @liggitt

Reply:
Yes, I believe @liggitt was handling the actual subpath validation changes, so wanted to make sure you guys agree upon the proper approach.

Reply:
https://github.com/kubernetes/kubernetes/pull/70490 is the direction we need to go: tolerate existing data on update, drop in PrepareForCreate/PrepareForUpdate rather than failing in validation

Reply:
#70490 is merged, please remove the feature gate checks from validation, and add them into podutil.DropDisabledFields, following the existing subpath example

Reply:
@liggitt merged in latest utils and changed validation to match subpath